library/core/src/array/mod.rs

```diff
@@ -426,6 +427,18 @@ impl<T: Clone, const N: usize> Clone for [T; N] {
     }
 }
 
+#[unstable(feature = "random", issue = "130703")]
+impl<T: Random, const N: usize> Random for [T; N] {
+    fn random(source: &mut (impl RandomSource + ?Sized)) -> Self {
+        let mut buf = [const { MaybeUninit::uninit() }; N];
+        for elem in &mut buf {
+            elem.write(T::random(source));
+        }
+        // SAFETY: all elements of the array were initialized.
+        unsafe { mem::transmute_copy(&buf) }
```

Comment:
Why not just use `array::from_fn`:
```suggestion
        array::from_fn(|_| T::random(source))
```

Reply:
I didn't know about this convenient function.